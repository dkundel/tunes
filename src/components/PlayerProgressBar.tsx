import * as React from 'react';

export interface PlayerProgressBarProps {
  percentage: number;
  title: string;
}

export default class PlayerProgressBar extends React.Component<
  PlayerProgressBarProps,
  undefined
> {
  public render() {
    return (
      <div className="player-progress" ref="progressContainer">
        <p className="player-title">{this.props.title}</p>
        <div
          className="player-progress-bar"
          ref="progressBar"
          style={{ '--percentage': this.props.percentage + '%' }}
        />
        <div
          className="player-title-overlay"
          ref="titleOverlay"
          style={{ '--percentage': this.props.percentage + '%' }}
        >
          <p className="player-title-overlay-content">
            {this.props.title}
          </p>
        </div>
      </div>
    );
  }
}
