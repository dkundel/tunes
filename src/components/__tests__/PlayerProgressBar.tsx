import * as React from 'react';
import PlayerProgressBar from '../PlayerProgressBar';
import * as renderer from 'react-test-renderer';

describe('test PlayerProgressBar component', () => {
  test('handles correct props', () => {
    const component = renderer.create(
      <PlayerProgressBar percentage={100} title="My Unit Test" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
