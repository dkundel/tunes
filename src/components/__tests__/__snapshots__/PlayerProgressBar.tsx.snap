// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PlayerProgressBar component handles correct props 1`] = `
<div
  className="sc-bdVaJa lgaJyD"
>
  <p
    className="sc-bwzfXH cXyljP"
  >
    My Unit Test
  </p>
  <div
    className="sc-bxivhb fvFpMe"
    style={
      Object {
        "--percentage": "100%",
      }
    }
  />
  <div
    className="sc-ifAKCX jeMYHp"
    style={
      Object {
        "--percentage": "100%",
      }
    }
  >
    <p
      className="sc-htpNat kCUXQt"
    >
      My Unit Test
    </p>
  </div>
</div>
`;
